### PR TITLE
Web API Documentation

### PR DESCRIPTION
**Pull request by michaeldeongreen**
* Updated API README.md file with latest instructions
* Implemented Swagger
* Modified API .gitignore file to ignore *.Development.json

### CODE REVIEW

File: api/src/LinkyLink/LinkyLink.csproj [hunk 1]
```diff
@@ -27,7 +27,11 @@
     <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
     <PackageReference Include="OpenGraph-Net" Version="3.2.0" />
     <PackageReference Include="Microsoft.AspNetCore.Cors" Version="2.2.0" />
+    <PackageReference Include="Swashbuckle.AspNetCore" Version="5.0.0-rc4" />
   </ItemGroup>
 
-
+  <PropertyGroup>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <NoWarn>$(NoWarn);1591</NoWarn>
```
Comment (wbreza) [inline api/src/LinkyLink/LinkyLink.csproj]: What is 1591 - We should consider adding a comment in here describing turning off this warning.

File: api/README.md [hunk 2]
```diff
@@ -1,80 +1,84 @@
...
 
 ## Build and run the backend locally
 
 ### Get the prerequisites
 
-- Install the [Azure Functions Core tools](https://docs.microsoft.com/azure/azure-functions/functions-run-local?WT.mc_id=theurlist-github-cephilli#install-the-azure-functions-core-tools) for your operating system
-- Install the [.NET Core SDK](https://dotnet.microsoft.com/download?WT.mc_id=theurlist-github-cephilli). This repo is pinned to use version 3.1.x of the SDK.
-- Install [Visual Studio Code](https://code.visualstudio.com/?WT.mc_id=theurlist-github-cephilli) or [Visual Studio Community edition](https://visualstudio.microsoft.com/vs?WT.mc_id=theurlist-github-cephilli)
+- Install the [.NET Core 3.1 SDK](https://dotnet.microsoft.com/download?WT.mc_id=theurlist-github-cephilli). This repo is pinned to use version 3.1.x of the SDK.
+- Install [Visual Studio](https://visualstudio.microsoft.com/) or [Visual Studio Code](https://code.visualstudio.com/?WT.mc_id=theurlist-github-cephilli) or [Visual Studio Community edition](https://visualstudio.microsoft.com/vs?WT.mc_id=theurlist-github-cephilli)
```
Comment (Humoiz) [inline api/README.md]: Is this link provided purposely? https://code.visualstudio.com/?WT.mc_id=theurlist-github-cephilli

File: api/src/LinkyLink/Controllers/LinksController.cs [hunk 3]
```diff
@@ -27,14 +27,18 @@ public LinksController(ILinksService linksService)
             _linksService = linksService;
         }
 
-        // Todo - Remove this method
-        // GET: api/Links
```
Comment (Humoiz) [inline api/src/LinkyLink/Controllers/LinksController.cs]: Can you delete this method.? Its not needed, I added that for testing, thats why there is a TODO to delete it later. I know its not in scope of this PR so skip it and I`ll delete it later.

File: api/src/LinkyLink/Startup.cs [hunk 4]
```diff
@@ -66,6 +89,13 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
             {
                 app.UseDeveloperExceptionPage();
             }
+            
+            //  Configure Swagger
```
Comment (Humoiz) [inline api/src/LinkyLink/Startup.cs]: Remove extra whitepace.

// Configure Swagger

File: api/theurlist_localhost_env.json [hunk 5]
```diff
@@ -4,7 +4,7 @@
 	"values": [
```
Comment (Humoiz) [inline api/theurlist_localhost_env.json]: Do we need this file anymore?

File: api/src/LinkyLink/Controllers/LinksController.cs (hunk 3 above)
Comment (michaeldeongreen) [reply api/src/LinkyLink/Controllers/LinksController.cs]: I can do it when I make changes to ValidationPage Controller.

File: api/theurlist_localhost_env.json (hunk 5 above)
Comment (michaeldeongreen) [reply api/theurlist_localhost_env.json]: Yes, even though they will be able to use Swagger, we want them to also be bale to use Postman, as the documentation already exists and I updated some of the screenshots as well.

File: api/src/LinkyLink/LinkyLink.csproj (hunk 1 above)
Comment (michaeldeongreen) [reply api/src/LinkyLink/LinkyLink.csproj]: Added Misc. Notes section in README.md and provided notes.

**Pull request merged**